test/SlowTests/Server/Documents/DataArchival/DataArchivalDataSubscriptionsTests.cs

```diff
@@ -491,5 +495,159 @@ public async Task DataSubscriptionTryoutResultsAreConsistentWithCurrentArchivedD
+            {
+                mre.Set();
+                Assert.True(await mre2.WaitAsync(reasonableWaitTime), "await mre2.WaitAsync(_reasonableWaitTime)");
+            });
+            
+            // wait for setting mre above
+            Assert.True(await mre.WaitAsync(reasonableWaitTime), "await mre.WaitAsync(_reasonableWaitTime)");
+            
+            
+            // drop subscription
+            await subscription.DisposeAsync(false);
+            try
+            {
+                // set mre2 and wait for fetchFromStorageTask to realize that its set already
+                mre2.Set();
+                await fetchFromStorageTask;
+            }
+            catch (Exception)
+            {
+                // no one cares
+            }
+            
+            Assert.True(exceptions.Count == 0, $"{string.Join(Environment.NewLine, exceptions.Select(x => x.ToString()))}");
+            
+            // assert that all previously fetched items are on the resend list already
+            var executor = store.GetRequestExecutor();
+            using var _ = executor.ContextPool.AllocateOperationContext(out var ctx);
+            var cmd = new GetSubscriptionResendListCommand(store.Database, subscription.SubscriptionName);
+            await executor.ExecuteAsync(cmd, ctx);
+            var res = cmd.Result;
+            
+            Assert.Equal(128, res.Results.Count);
+            
+            // Activate the archival manually - change docs archival status while they're on the resend list
+            await SetupDataArchival(store);
+            var database = await Databases.GetDocumentDatabaseInstanceFor(store);
+            database.Time.UtcDateTime = () => DateTime.UtcNow.AddMinutes(10);
+            var documentsArchiver = database.DataArchivist;
+            await documentsArchiver.ArchiveDocs();
+            
+            // start new worker it will process from resend
+            subscription = store.Subscriptions.GetSubscriptionWorker<User>(new SubscriptionWorkerOptions(sub)
+            {
+                TimeToWaitBeforeConnectionRetry = TimeSpan.FromMilliseconds(16),
+                MaxDocsPerBatch = 1
+            });
+            try
+            {
+                var items = new HashSet<string>();
+                var fetchFromResendListTask = subscription.Run(batch =>
+                {
+                    foreach (var item in batch.Items)
+                    {
+                        items.Add(item.Id);
+                    }
+                });
+                
+                // assert that documents are skipped in this scenario
+                Assert.Equal(0, await WaitForValueAsync(() => items.Count, docs, timeout: 10_000));
+            }
+            finally
+            {
+                await subscription.DisposeAsync();
+            }
+        }
+    }
+    private class GetSubscriptionResendListCommand : RavenCommand<ResendListResult>
```

Comment:
can we reuse the command from `ConcurrentSubscriptionsTests.cs` ?

https://github.com/ravendb/ravendb/blob/ad8970093ed5ed2d4d1fcb7e035335b506aaa48f/test/SlowTests/Client/Subscriptions/ConcurrentSubscriptionsTests.cs#L1012

Reply:
done